File: CS/FilterReportParameter.Module/BusinessObjects/TestClass.cs
using System;
using System.Linq;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using System.Collections.Generic;
using DevExpress.Persistent.BaseImpl;

namespace FilterReportParameter.Module.BusinessObjects {
    [DefaultClassOptions]
    public class DemoClass : BaseObject {
        public DemoClass(Session session)
            : base(session) {
        }
        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { SetPropertyValue("Name", value); }
        }
        public DemoParameter Parameter {
            get { return GetPropertyValue<DemoParameter>("Parameter"); }
            set { SetPropertyValue("Parameter", value); }
        }
    }
}

File: VB/FilterReportParameter.Module/BusinessObjects/TestClass.vb
Imports System
Imports System.Linq
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports System.Collections.Generic
Imports DevExpress.Persistent.BaseImpl

Namespace FilterReportParameter.Module.BusinessObjects
    <DefaultClassOptions> _
    Public Class DemoClass
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Property Name() As String
            Get
                Return GetPropertyValue(Of String)("Name")
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Name", value)
            End Set
        End Property
        Public Property Parameter() As DemoParameter
            Get
                Return GetPropertyValue(Of DemoParameter)("Parameter")
            End Get
            Set(ByVal value As DemoParameter)
                SetPropertyValue("Parameter", value)
            End Set
        End Property
    End Class
End Namespace
